### PR TITLE
Issue 10: Added the search text functionality

### PR DESCRIPTION
**Pull request by MrSriHarsha**

### CODE REVIEW

File: Checklist/ChecklistTableViewController.swift [hunk 1]
```diff
@@ -89,10 +97,52 @@ class ChecklistTableViewController: UITableViewController {
             let attributeString: NSMutableAttributedString =  NSMutableAttributedString(string: cell.textEnteredLabel.text!)
             attributeString.addAttribute(NSStrikethroughStyleAttributeName, value: 2, range: NSMakeRange(0, attributeString.length))
             cell.textEnteredLabel.attributedText = attributeString
+        } else if shouldShowSearchResults {
+            cell.textEnteredLabel.text = filteredChecklists[indexPath.row].text
         } else {
-            cell.textEnteredLabel.text = checklists[indexPath.row].text
+            cell.textEnteredLabel.text = checklist.text
```
Comment (sgoodwin) [inline Checklist/ChecklistTableViewController.swift]: I think your logic here might be a bit backwards. First you check if a regular checklist item is done and then you check if you should show search results if it's not?

File: Checklist/ChecklistTableViewController.swift [hunk 2]
```diff
@@ -89,10 +97,52 @@ class ChecklistTableViewController: UITableViewController {
...
+        tableView.reloadData()
+    }
+    
+    func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        shouldShowSearchResults = false
+        tableView.reloadData()
+    }
+    
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        if !shouldShowSearchResults {
```
Comment (sgoodwin) [inline Checklist/ChecklistTableViewController.swift]: This check isn't necessary, is it?

File: Checklist/ChecklistTableViewController.swift [hunk 3]
```diff
@@ -89,10 +97,52 @@ class ChecklistTableViewController: UITableViewController {
...
+            shouldShowSearchResults = true
+            tableView.reloadData()
+        }
+        searchController.searchBar.resignFirstResponder()
+    }
+    
+    func updateSearchResultsForSearchController(searchController: UISearchController) {
+        let searchString = searchController.searchBar.text
+        filteredChecklists = checklists.filter({ (checklist: Checklist) -> Bool in
+            return checklist.text.lowercased().contains(searchString!.lowercased())
```
Comment (sgoodwin) [inline Checklist/ChecklistTableViewController.swift]: 👍

File: Checklist/ChecklistTableViewController.swift (hunk 1 above)
Comment (MrSriHarsha) [inline Checklist/ChecklistTableViewController.swift]: method

Here I'm getting the checklist from the checklists array. I'm assigning the checklist text to the label. Haven't changed any functionality.

Comment (sgoodwin) [inline Checklist/ChecklistTableViewController.swift]: If you're showing filtered results, that first line will execute first. Shouldn't we only be looking in the ```checklists``` array if we *aren't* showing filtered results?

**Pull request merged**